<launch>

  <arg name="config_file" default="$(find toy_map_viewer)/config/realtime.yaml"/>

  <node pkg="rf_tf_broadcaster" type="map_tf_broadcaster_node" name="map_tf_broadcaster" output="screen"/>

  <arg name="date" default="2025-09-26-14-21-28_maxen_v6_2" />

  <node name="pointcloud_viewer" pkg="toy_map_viewer" type="pointcloud_viewer_node" output="screen">
    <rosparam file="$(arg config_file)" command="load"/>
  </node>

  <node name="map_to_global_viewer" pkg="toy_map_viewer" type="map_to_global_viewer_node" output="screen">
    <rosparam file="$(arg config_file)" command="load"/>
  </node>
  
  <!-- <node pkg="rosbag" type="play" name="sample_pc" args="-k $(find toy_map_viewer)/bag/kusn95_after.bag"/> -->
  
  <node pkg="toy_map_viewer" type="realtime_builder_node" name="realtime_builder_node" output="screen">
    <rosparam file="$(arg config_file)" command="load"/>
  </node>

  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find toy_map_viewer)/rviz/realtime_viewer.rviz" required="false" />
</launch>